domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -215,28 +498,26 @@ class TopicListController @Inject constructor(
               nextChapterSummary.name,
               nextChapterSummary.explorationId
             )
-            if (numberOfDaysPassed < ONE_WEEK_IN_DAYS) {
-              ongoingStoryListBuilder.addRecentStory(promotedStory)
-            } else {
-              ongoingStoryListBuilder.addOlderStory(promotedStory)
-            }
+            recommendedStories.add(promotedStory)
           }
         }
       }
     }
-    if ((ongoingStoryListBuilder.olderStoryCount + ongoingStoryListBuilder.recentStoryCount) == 0) {
-      ongoingStoryListBuilder.addAllRecentStory(createRecommendedStoryList())
-    }
-    return ongoingStoryListBuilder.build()
-  }
 
-  private fun createRecommendedStoryList(): List<PromotedStory> {
-    val recommendedStories = ArrayList<PromotedStory>()
     val topicIdJsonArray = jsonAssetRetriever
       .loadJsonFromAsset("topics.json")!!
       .getJSONArray("topic_id_list")
+
+    val topicList = ArrayList<String>()
     for (i in 0 until topicIdJsonArray.length()) {
-      recommendedStories.add(createRecommendedStoryFromAssets(topicIdJsonArray[i].toString()))
+      topicList.add(topicIdJsonArray[i].toString())
+    }
+
+    val index = topicList.indexOf(topicProgressList[topicProgressList.size - 1].topicId)
```

Comment:
More Kotlin idiomatic way to do this would be: ``val index = topicList.indexOf(topicProgressList.last().topicId)``

Reply:
Done.